Sources/SwiftGenCLI/templates/strings/flat-swift5.stencil

```diff
@@ -71,6 +71,16 @@ import Foundation
 // MARK: - Implementation Details
 
 extension {{enumName}} {
+  #if swift(>=5.6)
+  private static func tr(_ table: String, _ key: String, _ args: any CVarArg..., fallback value: String) -> String {
+    {% if param.lookupFunction %}
+    let format = {{ param.lookupFunction }}(key, table, value)
+    {% else %}
+    let format = {{param.bundle|default:"BundleToken.bundle"}}.localizedString(forKey: key, value: value, table: table)
+    {% endif %}
+    return String(format: format, locale: Locale.current, arguments: args)
+  }
+  #else
```

Comment:
Is it not creating a bit too much duplicated code to maintain here?

Cannot we just limit the `#if` to the `func` line?

Something more like this?

```suggestion
  #if swift(>=5.6)
  private static func tr(_ table: String, _ key: String, _ args: any CVarArg..., fallback value: String) -> String {
  #else
  private static func tr(_ table: String, _ key: String, _ args: CVarArg..., fallback value: String) -> String {
  #endif
```

Reply:
I agree with you that there is some strain on its maintenance. However, such a syntax is invalid in Swift.

![](https://github.com/SwiftGen/SwiftGen/assets/13805382/7ea85473-7e81-46bc-998a-c6aa7a6675ff)